src/sentry/static/sentry/app/views/settings/components/tag.tsx

```diff
@@ -1,22 +1,27 @@
-import PropTypes from 'prop-types';
 import React from 'react';
 import styled from '@emotion/styled';
 
 import InlineSvg from 'app/components/inlineSvg';
+import {Theme} from 'app/utils/theme';
 
-const getMarginLeft = p => {
-  if (!p.inline) {
-    return '';
-  }
-  return `margin-left: ${p.size === 'small' ? '0.25em' : '0.5em'};`;
+type Props = React.HTMLAttributes<HTMLDivElement> & {
+  priority?: keyof Theme['badge'] | keyof Theme['alert'];
+  size?: string;
+  icon?: string;
+  border?: boolean;
+  inline?: boolean;
 };
 
-const getBorder = p =>
-  p.border
-    ? `border: 1px solid ${
-        p.priority ? p.theme.alert[p.priority].border : p.theme.gray1
-      };`
-    : '';
+type StyleFucProps = Props & {theme: Theme};
+
+const getMarginLeft = (p: StyleFucProps) =>
+  p.inline ? `margin-left: ${p.size === 'small' ? '0.25em' : '0.5em'};` : '';
+
+const getPriority = (p: StyleFucProps) =>
+  p.priority ? p.theme.alert[p.priority] ?? p.theme.badge[p.priority] ?? null : null;
```

Comment:
this is kind of hard to parse

Reply:
Yeah I made it a bit more obvious and added a comment